test/library/packages/LinearAlgebra/correctness/no-dependencies/testExpm.chpl

```diff
@@ -0,0 +1,459 @@
+use LinearAlgebra;
+use TestUtils;
+
+{
+  const vec = 0..1;
+  var D = {vec,vec};
+  var X = eye(D);
+  X(0,1) = 1;
+  X = reshape(2*X, D);
+
+  var E = expm(X);
+
+  X(0,0) = 7.389056;
+  X(0,1) = 14.778112;
+  X(1,0) = 0.0;
+  X(1,1) = 7.389056;
+
+  assertAlmostEqual(E, X, 'Matrix Exponential - X = 2*2');
+}
+
+{
+  const vec = 0..2;
+  var D = {vec,vec};
+  var X = eye(D);
+  X(0,1) = 1;
+  X(0,2) = 1;
+  X(2,0) = 1;
+  X = reshape(2*X, D);
+
+  var E = expm(X);
+
+  X(0,0) = 27.799075;
+  X(0,1) = 26.799075;
+  X(0,2) = 26.799075;
+  X(1,0) = 0.0;
+  X(1,1) = 7.389056;
+  X(1,2) = 0.0;
+  X(2,0) = 26.799075;
+  X(2,1) = 20.410019;
+  X(2,2) = 27.799075;
+
+  assertAlmostEqual(E, X, 'Matrix Exponential - X = 3*3');
+}
+
+{
```

Comment:
For these diagonal matrices, you should use `setDiag`. Again, mostly cleaner to read (and the intention is more obvious).